src/bscPlugin/validation/ScopeValidator.ts

```diff
@@ -60,9 +60,10 @@ export class ScopeValidator {
         if (this.event.program.globalScope === this.event.scope) {
             return;
         }
-        this.event.program.diagnostics.clearByFilter({ scope: this.event.scope, tag: ScopeValidatorDiagnosticTag });
+        this.event.program.diagnostics.clearByFilter({ scope: this.event.scope, segment: null, tag: ScopeValidatorDiagnosticTag });
```

Comment:
Setting these to null is a bit unintuitive, especially combined with the `!== undefined` above. Is this intentional? Throughout the codebase, we've tried to basically treat `null` and `undefined` as the same thing. 

What does setting it to `null` accomplish? Does it say "don't match on segements"? Or "mach all segments"? I'm wondering if there's a more clear way to handle this, since plugin authors will also be calling these functions.